src/PoshGitTypes.ps1

```diff
@@ -144,7 +150,47 @@ class PoshGitTextSpan {
                 $fg = Get-ForegroundVirtualTerminalSequence $fg
             }
 
-            $str = "${fg}${bg}${txt}${e}0m"
+            if (($null -ne $fg) -or ($null -ne $bg)) {
+                $str = "${fg}${bg}${txt}${e}0m"
+            }
+            else {
+                $str = $txt
+            }
+        }
+
+        return $str
+    }
+
+    [string] ToEscapedString() {
+        if ($global:GitPromptSettings.AnsiConsole) {
```

Comment:
Not a blocker, but should `ToEscapedString()` care if the console supports ANSI? Properly encoding/escaping a string doesn't require it…